x/mint/types/inflations_test.go

```diff
@@ -0,0 +1,181 @@
+	"math"
+	"math/rand"
+	"strconv"
+	"testing"
+)
+
+func resetSupply(app *simapp.SimApp, ctx sdk.Context, initSupply sdk.Coins, curSupply sdk.Coins) {
+	err := app.BankKeeper.MintCoins(ctx, types.ModuleName, initSupply)
+	if err != nil {
+		panic(err)
+	}
+	err = app.BankKeeper.BurnCoins(ctx, types.ModuleName, curSupply)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getTestingAccounts(r *rand.Rand, n int, ctx sdk.Context, app *simapp.SimApp) []simtypes.Account {
+	accounts := simtypes.RandomAccounts(r, n)
+
+	for _, account := range accounts {
+		acc := app.AccountKeeper.NewAccountWithAddress(ctx, account.Address)
+		app.AccountKeeper.SetAccount(ctx, acc)
+	}
+
+	return accounts
+}
+
+func TestHandleInflations(t *testing.T) {
+	app := simapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+	acc := auth.NewEmptyModuleAccount(types.ModuleName, auth.Minter, auth.Burner)
+	app.AccountKeeper.SetModuleAccount(ctx, acc)
+
+	s := rand.NewSource(1)
+	r := rand.New(s)
+	targetAccounts := getTestingAccounts(r, 3, ctx, app)
+
+	minter := types.DefaultInitialMinter()
+	params := types.DefaultParams()
+
+	app.MintKeeper.SetParams(ctx, params)
+	app.MintKeeper.SetMinter(ctx, minter)
+
+	testDenom := "testDenom"
+	initSupplyAmount, _ := sdk.NewIntFromString("1000000000000000000")
+	initSupplyCoin := sdk.NewCoin(testDenom, initSupplyAmount)
+
+	tests := []struct {
+		inflation       types.Inflation
+		expectedBalance sdk.Int
+	}{
+		{types.NewInflation(testDenom, targetAccounts[0].Address.String(), sdk.NewDecWithPrec(1, 2)), sdk.NewInt(1576534791)},
+		{types.NewInflation(testDenom, targetAccounts[1].Address.String(), sdk.NewDecWithPrec(2, 2)), sdk.NewInt(3137536969)},
+		{types.NewInflation(testDenom, targetAccounts[2].Address.String(), sdk.NewDecWithPrec(3, 2)), sdk.NewInt(4683309617)},
+	}
+	for _, tc := range tests {
+		minter.Inflations = []*types.Inflation{&tc.inflation}
+		// Mint initial supply
+		resetSupply(app, ctx, sdk.NewCoins(initSupplyCoin), sdk.NewCoins(app.BankKeeper.GetSupply(ctx, testDenom)))
+
+		// Calculate inflation
+		inflationRatePerBlock, err := types.CalculateInflationPerBlock(&tc.inflation, params.BlocksPerYear)
+		if err != nil {
+			panic(err)
+		}
+
+		newCoinsToSend := types.CalculateInflationNewCoins(inflationRatePerBlock, app.BankKeeper.GetSupply(ctx, testDenom))
+		require.Equal(t, tc.expectedBalance.Int64(), newCoinsToSend.AmountOf(testDenom).Int64())
+
+		// Mint inflation returns
+		require.NoError(t, app.BankKeeper.MintCoins(ctx, types.ModuleName, newCoinsToSend))
+
+		// Convert targetAddress to sdk.AccAddress
+		acc, err := sdk.AccAddressFromBech32(tc.inflation.TargetAddress)
+		require.NoError(t, err)
+
+		// Send new reward tokens to target address
+		err = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, acc, newCoinsToSend)
+		require.NoError(t, err)
+
+		// Assert tokens reached account
+		testAccountBalance := app.BankKeeper.GetBalance(ctx, acc, tc.inflation.Denom)
```

Comment:
Shouldn't we call here `mint.HandleInflations(...)` function rather than all lines I merged above? If I understand it correctly, that is the point/objective of this test.